### PR TITLE
Fix payload size check in HostName module

### PR DESCRIPTION
**Pull request by simonjaeger**
## Description

Fixed a check that failed in the *HostName* module on *MmiSet* when a session was created with a maximum payload size of zero. Corrected by skipping the check if the maximum payload size is zero and created a test to verify fix. 

## Checklist

- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-osconfig/blob/main/CONTRIBUTING.md).
- [x] I added unit-tests to validate my changes. All unit tests are passing.
- [x] I have merged the latest `main` branch prior to this PR submission.
- [x] I submitted this PR against the `main` branch.

### CODE REVIEW

**Review by MariusNi — APPROVED** [review]
:shipit:

File: src/modules/hostname/tests/HostNameBaseTests.cpp [hunk 1]
```diff
@@ -118,6 +118,27 @@ namespace OSConfig::Platform::Tests
...
+
+        HostNameBaseTests testModule(textResults, 0);
+        int status = testModule.Get(&testModule, HostNameBase::m_componentName, HostNameBase::m_propertyName, &payload, &payloadSizeBytes);
+
+        std::string result(payload, payloadSizeBytes);
+
+        EXPECT_EQ(status, MMI_OK);
+        EXPECT_STREQ(result.c_str(), "\"device\"");
+
+        delete payload;
```
Comment (robertschaedler3) [inline src/modules/hostname/tests/HostNameBaseTests.cpp:139]: nit: use `FREE_MEMORY()` macro from common utils for consistency

**Review by robertschaedler3 — DISMISSED** [review]
👍🏻

File: src/modules/hostname/tests/HostNameBaseTests.cpp (hunk 1 above)
Comment (simonjaeger) [reply src/modules/hostname/tests/HostNameBaseTests.cpp]: Good point, thanks! Fixed.

File: src/modules/hostname/tests/HostNameBaseTests.cpp [hunk 2]
```diff
@@ -118,6 +118,27 @@ namespace OSConfig::Platform::Tests
         delete payload;
```
Comment (MariusNi) [inline src/modules/hostname/tests/HostNameBaseTests.cpp:118]: >delete payload; [](http://example.com/codeflow?start=8&length=15)

Apply same as on line 139 everywhere else now.

**Comment by MariusNi** [comment]
>         int payloadSizeBytes;

nit, for consistency with rest of project. always explicitly initialize variables. Here:

MMI_JSON_STRING payload = nullptr;
int payloadSizeBytes = 0;

---
Refers to: src/modules/hostname/tests/HostNameBaseTests.cpp:157 in bffd8a5. [](commit_id = bffd8a5ae8f2546bc46370c01f3dc526611fc25a, deletion_comment = False)

**Comment by MariusNi** [comment]
>         int payloadSizeBytes;

In all instances in this file, please.

---
In reply to: [1258619152](https://github.com/Azure/azure-osconfig/pull/399#issuecomment-1258619152) [](http://example.com/codeflow?ancestors=1258619152)

---
Refers to: src/modules/hostname/tests/HostNameBaseTests.cpp:157 in bffd8a5. [](commit_id = bffd8a5ae8f2546bc46370c01f3dc526611fc25a, deletion_comment = False)

File: src/modules/hostname/tests/HostNameBaseTests.cpp (hunk 2 above)
Comment (simonjaeger) [reply src/modules/hostname/tests/HostNameBaseTests.cpp]: Good point, thanks!

**Pull request merged**